lxd/instance/drivers/driver_lxc.go

```diff
@@ -8188,6 +8188,31 @@ func (d *lxc) CGroup() (*cgroup.CGroup, error) {
 	return d.cgroup(cc, true)
 }
 
+// SetAffinity sets affinity in the container according with a set provided.
+func (d *lxc) SetAffinity(set []string) error {
+	sort.Strings(set)
+	affinitySet := strings.Join(set, ",")
+
+	// Confirm the container didn't just stop
+	if d.InitPID() <= 0 {
+		return nil
+	}
+
+	cg, err := d.CGroup()
+	if err != nil {
+		d.logger.Error("balance: Unable to get cgroup struct", logger.Ctx{"name": d.Name(), "err": err, "value": affinitySet})
+		return err
```

Comment:
Lets return a wrapped error here like:

```go
return fmt.Errorf("Unable to get cgroup struct: %w", err)
```

The `set` can be logged in the caller as its passed in. No need to log here.

Reply:
Then get rid of the logger from this function.

Reply:
Fixed, thanks!